### ISSUE TITLE
Pretrain GIN models using UnlabeledSmiles

### ISSUE THREAD

**Issue by skrsna** [awslabs/dgl-lifesci#90]
Hi, 

I know that one can use pre-trained GIN models with masked and contextpred modes etc for downstream regression or classification. But I'm wondering if there are any training scripts or resources in dgl-life to pretrain GIN models with different node/edge features. If not I'm planning to implement it myself and I can open a PR here. 

Thanks!!

**Comment by mufeili** [comment]
Hi,

DGL-LifeSci has not supported pre-training GIN models with different node/edge features. If you can make a contribution for that, that will be great!

**Comment by skrsna** [comment]
Thanks! If you can point me to training scripts you used to generate pre-trained models with the current node/edge features for GIN that'd be great. If not I can start writing them based of the PyG implementation from here (https://github.com/snap-stanford/pretrain-gnns). Also please feel free to close the issue. I'll open a PR once I have the implementation. Cheers!!

**Comment by mufeili** [comment]
> Thanks! If you can point me to training scripts you used to generate pre-trained models with the current node/edge features for GIN that'd be great. If not I can start writing them based of the PyG implementation from here (https://github.com/snap-stanford/pretrain-gnns). Also please feel free to close the issue. I'll open a PR once I have the implementation. Cheers!!

I directly adopted the pre-trained weights released by the authors, so there isn't any DGL-based training script I can share. Let me know if you need any help in adapting the authors' code. Thanks a lot!

**Comment by skrsna** [comment]
Great! I'll open a work in progress PR and keep updating the scripts so that you can keep track and give suggestions!  I'll close the issue for now.
